### PR TITLE
feat(train): automatically decide batch_size

### PR DESCRIPTION
**Pull request by 34j**

### CODE REVIEW

**Comment by soliloqueen** [comment]
Doesn't batch size determine more than speed? Like if you increase it, the accuracy can go down

**Comment by markrmiller** [comment]
You can make up for model performance loss by also adjusting the learning rate. You may still get weaker generalization though.

But I'd bet if you are already using at least 8, and you generally want a multiple of 8, the difference between 8/16/32 here is not going to be wild with no other changes.
